### DISCUSSION TITLE
WebAssembly support?

### DISCUSSION THREAD

**Discussion by vdawg-git** [kas-gui/tutorials#20] [Q&A] [answered]
Hey there, does this library support being compiled to WebAssembly and to then be used alongside a regular DOM-based UI?

**Comment by dhardy** [comment, marked as answer]
Nope. That might be a nice goal eventually, but (a) the current model doesn't support DOM at all (it would be interesting to see whether DOM could be used as a renderer in lieu of `kas-wgpu`) and (b) WASM isn't tested (last time I tried was a failure, possibly because of WGPU).

**Reply by vdawg-git** [reply]
Ah I see, damn
I meant to render the Kas-UI via a canvas in the browser for a performance-heavy UI and then some JS UI on top for "simple" stuff :D 

Thank you for the reply :)

**Reply by dhardy** [reply]
`kas-wgpu` might work in a canvas... but I guess some fixes are required.

In any case, KAS is not what I'd call production quality or stable yet.
